src/app/story/directives/status.component.ts

```diff
@@ -67,119 +46,73 @@ export class StoryStatusComponent implements DoCheck {
     }
   }
 
-  formatInvalid(str: string): string {
-    str = str.trim();
-    str = str.replace(/shortdescription/i, 'teaser');
-    str = str.charAt(0).toUpperCase() + str.slice(1);
-    return str;
-  }
-
-  formatInvalids(strict = false): string[] {
-    let invalids = strict ? this.strictInvalid : this.normalInvalid;
-    if (invalids) {
-      return invalids.split(',').map(s => this.formatInvalid(s));
-    } else {
-      return [];
+  get date() {
+    if (this.story && this.story.releasedAt) {
+      return this.story.releasedAt;
+    } else if (this.story && this.story.publishedAt && this.story.publishedAt.valueOf() <= new Date().valueOf()) {
+      return this.story.publishedAt;
     }
   }
 
-  countProblems(strict = false): string {
-    let count = this.formatInvalids(strict).length;
-    return count === 1 ? `Found 1 problem` : `Found ${count} problems`;
-  }
-
-  showProblems() {
-    let normals = this.formatInvalids(false);
-    let stricts = this.formatInvalids(true).filter(s => normals.indexOf(s) === -1);
-
-    let title = 'Validation errors';
-    let msg = '';
-    normals.forEach(s => msg += `<li class="error">${s}</li>`);
-    if (this.isPublished) {
-      stricts.forEach(s => msg += `<li class="error">${s}</li>`);
-    }
-    if (this.id && !this.isPublished) {
-      stricts.forEach(s => msg += `<li>${s}</li>`);
-      if (normals.length === 0) {
-        title = 'Not ready to publish';
+  setDate(date: Date) {
+    if (this.story) {
+      // set releasedAt, see CMS story model update_published_to_released
+      this.story.set('releasedAt', date);
+      // Legacy: This actually does nothing. Removing date from datepicker is not valid and does not emit change.
+      // ...but if it did actually unset releasedAt, then we'd want to notify so keep
+      if (this.nextStatus !== 'draft' && !date) {
+        this.notifyOfCanceledPublication();
       }
     }
-
-    this.modal.show({title: title, body: `<ul>${msg}</ul>`, secondaryButton: 'Okay'});
   }
 
-  toggleEdit() {
-    this.editStatus = !this.editStatus;
+  get dateChanged(): boolean {
+    return this.story && this.story.changed('releasedAt', false);
   }
 
-  togglePublish() {
-    this.isPublishing = true;
-    this.story.setPublished(!this.story.publishedAt).subscribe(() => {
-      this.angulartics2.eventTrack.next({ action: this.story.publishedAt ? 'publish' : 'unpublish',
-        properties: { category: 'episode', label: 'episode/' + this.story.doc.id }});
-      this.toastr.success(`Episode ${this.story.publishedAt ? 'published' : 'unpublished'}`);
-      this.isPublishing = false;
-      this.editStatus = false;
-    });
+  notifyOfCanceledPublication() {
+    const futurePublished = this.story.publishedAt && new Date() < this.story.publishedAt;
+    const removingReleaseDate = this.story.changed('releasedAt') && !this.story.releasedAt;
+    if (removingReleaseDate && futurePublished) {
+      this.modal.alert(
```

Comment:
This is very helpful.

Reply:
😬 so this doesnt actually alert anymore because I found users can't remove the date (it's not valid and doesn't emit a change) but I kept it because _if_ things changed to where they could clear the date field and that was valid, we'd want to make sure they knew what would happen. It used to work with the old checkbox, so maybe we still need it given these new thoughts?

Reply:
Oh, sorry, I was thinking of the "You do not need to unpublish the episode to update the content" confirmation modal, not this one!

But I'm fine with leaving this in here, just in case we do get a datepicker field someday that allows clearing and emits null/blank.